### ISSUE TITLE
CI is failing on master, but building fine

### ISSUE THREAD

**Issue by brianamarie** [github/training-kit#729]
## Overview
**TL;DR**

CI is failing on master, but the page is still building and deploying fine. I don't know what's causing the build failure. 

cc @a-a-ron for comms duty, but I'll also try and find time to poke around with this and see if I can fix it. 

#### Screenshots
[You can see the error message here](https://github.com/github/training-kit/commit/cec2b5947f54b230b1c4cd2f89b8db9ac7965c5b/checks?check_suite_id=423792222).

### How to Contribute
- [ ] Comment on this issue if you'd like to work/collaborate on it!
- [ ] Fork the [repository](https://github.com/github/training-kit).
- [ ] Use the [GitHub Flow](https://guides.github.com/introduction/flow/) to make changes to your fork. [This](https://services.github.com/on-demand/intro-to-github/) is a refresher course if you're unsure about how to make a change on GitHub.
- [ ] Push your changes to your repository.
- [ ] Submit a Pull Request
  - Base Dropdown: github/training-kit
  - Compare Dropdown: Your fork

#### Testing Locally
If you'd like to make and test changes locally (and see how they would look if merged), do the following:
- In your command line:
  - [Install Git](https://git-scm.com/)
  - [Install Ruby](https://www.ruby-lang.org/en/documentation/installation/)
  - [ ] Run `script/setup`
  - [ ] Run `script/server`.
    - When successful, the script will initiate a local server at `http://127.0.0.1:4000/`.

### Questions?
- Leave a comment on this issue! Make sure to use @ mentions if you want a specific person's attention!

**Comment by NishkarshRaj** [comment]
@brianamarie I am an expert on CI except Jenkins and basic Actions but I will give this a shot.
Also, on the attached screenshot error message in Node module shows that
```
Cannot find module 'semver'
```
Is 'semver' a thing or the script has spelled server wrong somewhere?

**Comment by NishkarshRaj** [comment]
I cannot get something I can work with but I have something substantial that the maintainers must look at. CC @brianamarie @a-a-ron 
So, while I was trying to find the root cause of the error in CI in **Actions** tab, I saw that CI fails after  #726 was merged but since it only updated a Markdown file, it must not affect the Continuous Integration part.

So, I looked at the #724 merge request which was the last stable and led to successful CI execution.

On closely analyzing its **Install node modules** [log](https://github.com/github/training-kit/runs/404794172?check_suite_focus=true), which causes recent errors, it can be seen that a warning message is generated stating that a lock file is created called `package-lock.json` and this file must be committed.

Maybe this lockfile is causing the problem?

**Comment by brianamarie** [comment]
@NishkarshRaj Thank you for taking the time to look into this 🙏 

I'm no expert in this, but everything that you're saying is making total sense. Next steps to me would be to try and recreate the same bug locally, then open a PR that adds a step for adding and committing the `package-lock.json` file in the build script, and opening a PR here.

What do you think? I'm happy to do that, but would also be happy to review your PR in case you want to keep working on this. 😊

**Comment by NishkarshRaj** [comment]
I will definitely try to solve this issue in my fork and raise a PR if I can @brianamarie 🙏 
But since I am also not experienced to handle such complex problem where Actions is using pre-built scripts and Docker, I think it would be better that some one more experienced handles this.

**Comment by NishkarshRaj** [comment]
@brianamarie Great work, I was unable to locate the package-lock file because of my inexperience in Docker and yml files but it feels good that I detected the issue correctly 😄
